com.reprezen.swagedit.tests/src/com/reprezen/swagedit/tests/ValidatorTest.xtend

```diff
@@ -187,4 +189,89 @@ class ValidatorTest {
 			assertEquals(IMarker.SEVERITY_ERROR, it.level)
 		]
 	}
-}
\ No newline at end of file
+
+	@Test
+	def void shouldWarnOnDuplicateKeys() {
+		val content = '''
+			swagger: '2.0'
+			swagger: '2.0'
+			info:
+			  version: 0.0.0
+			  title: Simple API
+			paths:
+			  /foo/{bar}:
+			    get:
+			      responses:
+			        '200':
+			          description: OK
+		'''
+
+		document.set(content)
+		val errors = validator.validate(document)
+
+		assertEquals(2, errors.size())
+		assertThat(errors, hasItems(
+			new SwaggerError(1, IMarker.SEVERITY_WARNING, String.format(Messages.error_duplicate_keys, "swagger")),
+			new SwaggerError(2, IMarker.SEVERITY_WARNING, String.format(Messages.error_duplicate_keys, "swagger"))
+		))
+	}
+
+	@Test
+	def void shouldWarnOnDuplicateKeys_InsideObjects() {
+		val content = '''
+			swagger: '2.0'
+			info:
+			  version: 0.0.0
+			  version: 1.0.0
+			  title: Simple API
+			paths:
+			  /foo/{bar}:
+			    get:
+			      responses:
+			        '200':
+			          description: OK
+		'''
+
+		document.set(content)
+		val errors = validator.validate(document)
+
+		assertEquals(2, errors.size())
+		assertThat(errors, hasItems(
+			new SwaggerError(3, IMarker.SEVERITY_WARNING, String.format(Messages.error_duplicate_keys, "version")),
+			new SwaggerError(4, IMarker.SEVERITY_WARNING, String.format(Messages.error_duplicate_keys, "version"))
+		))
+	}
+
+	@Test
+	def void shouldWarnOnDuplicateKeys_InsideItems() {
+		val content = '''
+			swagger: '2.0'
+			info:
+			  version: 0.0.0
+			  title: Simple API
+			paths:
+			  /foo/{bar}:
+			    get:
+			      parameters:
+			         - name: bar
+			           in: path
+			           in: path
+			           type: string
+			           required: true
+			      responses:
+			        '200':
+			          description: OK
+		'''
+
+		document.set(content)
+		val errors = validator.validate(document)
+
+		assertEquals(2, errors.size())
+		assertThat(errors, hasItems(
+			new SwaggerError(10, IMarker.SEVERITY_WARNING, String.format(Messages.error_duplicate_keys, "in")),
+			new SwaggerError(11, IMarker.SEVERITY_WARNING, String.format(Messages.error_duplicate_keys, "in"))
+		))
+	}
```

Comment:
I would also add a test for a case when the value is an object, for example here `responses` defined twice within the same pathItem:
```yaml
swagger: '2.0'
info:
  version: 0.0.0
  title: Simple API
paths:
  /foo/{bar}:
    get:
      responses:
        '200':
          description: OK
      responses:
        '200':
          description: OK
```
It works as expected - validator shows a warning, but we don't have a test for it.